cdap-common/src/main/java/co/cask/cdap/common/security/AuthEnforceRewriter.java

```diff
@@ -242,11 +261,14 @@ public void visitEnd() {
             return;
           }
           AuthEnforceAnnotationNodeProcessor nodeProcessor =
-            new AuthEnforceAnnotationNodeProcessor(authEnforceAnnotationNode);
-          Map<String, Integer> paramAnnotation = processParameterAnnotationNode(parameterAnnotationNode);
+                  new AuthEnforceAnnotationNodeProcessor(authEnforceAnnotationNode);
+          Map<String, Integer> paramPositions = processParameterAnnotationNode(parameterDetails);
 
           List<EntityPartDetail> entityPartDetails = new ArrayList<>();
           for (String name : nodeProcessor.getEntities()) {
+            // just need to check for empty string since it cannot be null as that will be a compilation error
+            Preconditions.checkArgument(!name.isEmpty(), "Found an empty string in entity parts of annotation %s on " +
```

Comment:
This check is not needed. If the name is indeed empty, there will be no field/param found to match it, hence will fail anyway in the logic down below.